CMakeLists.txt

```diff
@@ -5,62 +5,88 @@ project(ni_grpc_device_server
+
+  set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
+  set(_REFLECTION grpc++_reflection)
+  set(_GRPC grpc)
+  set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
+  set(_GRPC_GRPCPP grpc++)
+  set(_PROTOBUF_LIBPROTOBUF libprotobuf)
+  set(_UTF8CPP utf8cpp)
+else()
   find_program(_PROTOBUF_PROTOC protoc)
   find_program(_GRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
+  find_library(_GRPC_GPR gpr)
+  find_library(_ABSEIL_SYNC absl_synchronization REQUIRED)
 
-  if(NOT _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
-    find_package(gRPC REQUIRED)
-    find_library(_REFLECTION grpc++_reflection)
-    find_library(_GRPC_GRPCPP grpc++)
-    find_library(_PROTOBUF_LIBPROTOBUF protobuf)
-  else()
+  if(USE_NILRT_LEGACY_TOOLCHAIN)
+    # The archetypical NILRT SDK toolchain build case
     add_subdirectory(third_party/grpc ${CMAKE_CURRENT_BINARY_DIR}/grpc EXCLUDE_FROM_ALL)
     set(_REFLECTION grpc++_reflection)
     set(_GRPC_GRPCPP grpc++)
     set(_PROTOBUF_LIBPROTOBUF libprotobuf)
-  endif()
+  else()
+    # The archetypical linux build case (including OpenEmbedded)
+    find_library(_GRPC_GRPCPP grpc++)
+    find_library(_PROTOBUF_LIBPROTOBUF protobuf)
+    find_library(_REFLECTION grpc++_reflection)
 
-else()
-  add_subdirectory(third_party/grpc ${CMAKE_CURRENT_BINARY_DIR}/grpc EXCLUDE_FROM_ALL)
-  set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
-  set(_REFLECTION grpc++_reflection)
-  set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
-  set(_GRPC_GRPCPP grpc++)
-  set(_PROTOBUF_LIBPROTOBUF libprotobuf)
-endif()
+    find_package(gRPC REQUIRED)
+    find_package(GTest REQUIRED)
+    find_package(nlohmann_json REQUIRED)
+    find_package(utf8cpp REQUIRED)
 
-#----------------------------------------------------------------------
-CreateVirtualEnvironment(virtual_environment
-  REQUIREMENTS_TXT
-    ${CMAKE_SOURCE_DIR}/python_build_requirements.txt
-  ENV_NAME
-    venv
-  OUT_PYTHON_EXE
-    PYTHON_EXE
-)
+    set(_GRPC gRPC::grpc)
+    set(_UTF8CPP utf8cpp::utf8cpp)
+  endif()
+endif()
 
-#----------------------------------------------------------------------
-# Use the utfcpp targets directly from this build, only when not cross-compiling.
-#----------------------------------------------------------------------
-if(CMAKE_CROSSCOMPILING AND NOT _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
-  find_package(utf8cpp REQUIRED)
+# Python3 Virtual Environment
+if(USE_PYTHON_VIRTUALENV)
+  include(CreateVirtualEnvironment)
+  CreateVirtualEnvironment(virtual_environment
+    REQUIREMENTS_TXT
+      ${CMAKE_SOURCE_DIR}/python_build_requirements.txt
+    ENV_NAME
+      venv
+    OUT_PYTHON_EXE
+      PYTHON_EXE
+  )
 else()
-  add_subdirectory(third_party/utfcpp ${CMAKE_CURRENT_BINARY_DIR}/utfcpp EXCLUDE_FROM_ALL)
+  find_package(Python3 REQUIRED)
+  set(PYTHON_EXE ${Python3_EXECUTABLE})
 endif()
-include_directories(
-  "./third_party/utfcpp/source"
-)
+
+enable_testing()
```

Comment:
I'm not familiar with these options. Can you add a comment on why they're here?

Reply:
This was my change. In truth, I don't fully understand why it is needed either. The [cmake documentation](https://cmake.org/cmake/help/latest/command/enable_testing.html) isn't very helpful. I move it here from later in the original file, where it was invoked only in the Windows build case. Since my changes are to homogenize the cmake pathways for all the build OSes, and to enable test execution on Linux Desktop and NILRT, I just moved this call into the main pathway.

Nothing I have seen about this command would lead me to believe that it is harmful in a linux build context.

Reply:
I am marking this comment as resolved for now.